rancher/rancher
# Backport 1.27 support to 2.7.11

## Issue: https://github.com/rancher/rancher/issues/44141

This is a back port of the following PRs 

+ https://github.com/rancher/rancher/pull/42808
+ https://github.com/rancher/rancher/pull/42953
+ https://github.com/rancher/rancher/pull/43229
+ https://github.com/rancher/rancher/pull/43081
+ https://github.com/rancher/rancher/pull/43178
+ https://github.com/rancher/rancher/pull/42994
+ https://github.com/rancher/rancher/pull/43685
+ https://github.com/rancher/rancher/pull/42605
https://github.com/rancher/rancher/pull/43258

`go generate` was also run in order to update the copyright date on all generated files

## Problem
2.7.11 needs support for k8s 1.27
 
## Solution
Various back ports 

## Details 

A majority of these commits are clean cherry picks from the above issues, however some commits have been omitted as they were redundant (e.g. https://github.com/rancher/rancher/pull/42808/commits/4accd494a559d34b5a67ef2385ac3f69023e5363 which would later be replaced by 20fe90ba2f98c39ab9fdabcf7e1e55f0634d4273). I can add those back into this PR if we really think they are necessary.

A few commits have not been cherry picked and were instead done by hand. The `go.mod` files in `2.8` differ significantly from those on `2.7.11` and creating new commits was less error prone than attempting to resolve all of the merge conflicts (d5e230fa45e57295126ad2a3ce5b4fb71554bf86, d2fdab5dac61e421991d830fa610f9678e92e254, 446cce0119fb3e83532a5c4620547575264cdfff)

I've also gone ahead and updated the charts branch from `dev-v2.7` to the newly created `dev-v2.7.11`, and updated kdm to point to `dev-v2.7`. 


Along with core k8s dependencies, the following rancher specific components have been updated 
+ rancher-machine, bumped from 100 to 103 in https://github.com/rancher/rancher/pull/42808
+ Wrangler, bumped to `v1.1.1-0.20230831050635-df1bd5aae9df` from `v1.1.1` in https://github.com/rancher/rancher/pull/42808
+ Norman, bumped to `v0.0.0-20230831160711-5de27f66385d` from `v0.0.0-20230426211126-d3552b018687` in https://github.com/rancher/rancher/pull/42808
+ Dynamic Listener, bumped to `v0.3.6` from `v0.3.5`
+ https://github.com/rancher/rancher/pull/42808  bumped RKE to `1.5.0-rc5` from `v1.5.0-rc2`, I have changed this to use `1.4.12` which is the latest release for use by Rancher 2.7.x 

 
## Testing
## Engineering Testing
### Manual Testing

I've run `make quick` locally and did not see any compilation errors, I ran the resulting docker image and saw that Rancher started up as expected.

I've also run Rancher in my IDE using Rancher Desktop running k3s `v1.27.4`. Rancher started as expected, and I was able to provision an RKE2 1.26 cluster on Digital Ocean. 


### Automated Testing
n/a 

## QA Testing Considerations
Should test all things related to 1.27 support, as well as revalidate  
+ https://github.com/rancher/rancher/pull/42953
+ https://github.com/rancher/rancher/pull/43229
+ https://github.com/rancher/rancher/pull/43081
+ https://github.com/rancher/rancher/issues

### Regressions Considerations
Refer to above PRs/issues on specific regression concerns
